darwin/Platform.c

```diff
@@ -114,33 +114,26 @@ const SignalItem Platform_signals[] = {
 
 const unsigned int Platform_numberOfSignals = ARRAYSIZE(Platform_signals);
 
+enum {
+   MEMORY_CLASS_WIRED = 0,
+   MEMORY_CLASS_SPECULATIVE,
+   MEMORY_CLASS_ACTIVE,
+   MEMORY_CLASS_PURGEABLE,
+   MEMORY_CLASS_COMPRESSED,
+   MEMORY_CLASS_INACTIVE,
+};
+
 const MemoryClass Platform_memoryClasses[] = {
-#define MEMORY_CLASS_WIRED       0
-   { .label = "wired",       .countsAsUsed = true,  .countsAsCache = false, .color = DYNAMIC_RED     }, // pages wired down to physical memory (kernel)
-#define MEMORY_CLASS_SPECULATIVE 1
-   { .label = "speculative", .countsAsUsed = true,  .countsAsCache = true,  .color = DYNAMIC_MAGENTA }, // readahead optimization caches
-#define MEMORY_CLASS_ACTIVE      2
-   { .label = "active",      .countsAsUsed = true,  .countsAsCache = false, .color = DYNAMIC_GREEN   }, // userland pages actively being used
-#define MEMORY_CLASS_PURGEABLE   3
-   { .label = "purgeable",   .countsAsUsed = false, .countsAsCache = true,  .color = DYNAMIC_YELLOW  }, // userland pages voluntarily marked "discardable" by apps
-#define MEMORY_CLASS_COMPRESSED  4
-   { .label = "compressed",  .countsAsUsed = true,  .countsAsCache = false, .color = DYNAMIC_BLUE    }, // userland pages being compressed (means memory pressure++)
-#define MEMORY_CLASS_INACTIVE    5
-   { .label = "inactive",    .countsAsUsed = true,  .countsAsCache = true,  .color = DYNAMIC_GRAY    }, // pages no longer used; macOS counts them as "used" anyway...
+   { .label = "wired",       .countsAsUsed = true,  .countsAsCache = false, .color = MEMORY_1 }, // pages wired down to physical memory (kernel)
+   { .label = "speculative", .countsAsUsed = true,  .countsAsCache = true,  .color = MEMORY_2 }, // readahead optimization caches
```

Comment:
If you're bothered to make an enum, it shouldn't be a problem to fix the table index like this:

```suggestion
   [MEMORY_CLASS_SPECULATIVE] = { .label = "speculative", .countsAsUsed = true,  .countsAsCache = true,  .color = MEMORY_2 }, // readahead optimization caches
```

Reply:
Especially using the array index notation is used elsewhere already. Makes things more readable.